src/Core/Services/RequestValidator.cs

```diff
@@ -428,17 +430,27 @@ public void ValidateUpsertRequestContext(UpsertRequestContext upsertRequestCtx)
         /// are valid for the provided request body.
         /// </summary>
         /// <param name="column">The column to be verified.</param>
+        /// <param name="exposedName">Exposed name of the column.</param>
         /// <param name="fieldsInRequestBody">The fields in the request body.</param>
+        /// <param name="primaryKey">The primary key of the source definition for the column.</param>
+        /// <param name="isReplacementUpdate">Indicates if the column is a replacement update.</param>
+        /// <param name="isRequestBodyStrict">Indicates if the runtime setting is request body strict.</param>
         /// <returns>true if the column is validated.</returns>
         private static bool ValidateColumn(ColumnDefinition column,
                                            string exposedName,
                                            IEnumerable<string> fieldsInRequestBody,
-                                           bool isReplacementUpdate)
+                                           IEnumerable<string> primaryKey,
+                                           bool isReplacementUpdate,
+                                           bool isRequestBodyStrict)
         {
             string message;
-            // Autogenerated values should never be present in a request body.
+            // Autogenerated values should never be present in a request body, unless they are both a primary key
+            // and request-body-strict is false.
             // TODO: Add a similar check for read-only fields.
-            if (column.IsAutoGenerated && fieldsInRequestBody.Contains(exposedName))
+            if (column.IsAutoGenerated &&
+                fieldsInRequestBody.Contains(exposedName) &&
+                (!primaryKey.Contains(exposedName) ||
```

Comment:
comparing primarykey with exposed name doesnt tell you whether the column is primary key or not.. exposed name could be anything...